src/Renderer/Device/Utils/DebugUtilsMessenger.cpp

```diff
@@ -2,54 +2,69 @@
 
 namespace SnekVk::DebugUtilsMessenger
 {
-          // local callback functions
-  static VKAPI_ATTR VkBool32 VKAPI_CALL debugCallback(
-      VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity,
-      VkDebugUtilsMessageTypeFlagsEXT messageType,
-      const VkDebugUtilsMessengerCallbackDataEXT *pCallbackData,
-      void *pUserData) {
-    std::cerr << "validation layer: " << pCallbackData->pMessage << std::endl;
-
-    return VK_FALSE;
-  }
-
-  VkResult CreateMessenger(
-      VkInstance instance,
-      const VkDebugUtilsMessengerCreateInfoEXT *pCreateInfo,
-      const VkAllocationCallbacks *pAllocator,
-      VkDebugUtilsMessengerEXT *pDebugMessenger) {
-    auto func = (PFN_vkCreateDebugUtilsMessengerEXT)vkGetInstanceProcAddr(
-        instance,
-        "vkCreateDebugUtilsMessengerEXT");
-    if (func != nullptr) {
-      return func(instance, pCreateInfo, pAllocator, pDebugMessenger);
-    } else {
-      return VK_ERROR_EXTENSION_NOT_PRESENT;
+
+    static VKAPI_ATTR VkBool32 VKAPI_CALL debugCallback(
+        VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity,
+        VkDebugUtilsMessageTypeFlagsEXT messageType,
+        const VkDebugUtilsMessengerCallbackDataEXT *pCallbackData,
+        void *pUserData) 
+    {
+        std::cerr << "validation layer: " << pCallbackData->pMessage << std::endl;
+
+        return VK_FALSE;
+    }
+
+    VkResult CreateMessenger(
+        VkInstance instance,
+        const VkDebugUtilsMessengerCreateInfoEXT *pCreateInfo,
+        const VkAllocationCallbacks *pAllocator,
+        VkDebugUtilsMessengerEXT *pDebugMessenger) 
+    {
+        // Vulkan does not load validation layer-related function by default. 
+        // vkGetIntstanceProcAddr allows us to query for a relevant function pointer
+        // and use that. We get the pointer, cast it to the type of function we want,
+        // then use it in the 'func' variable. 
+
+        // In our case, we're looking 'vkCreateDebugUtilsMessengerEXT' so we can create
+        // our debug messenger.
```

Comment:
I like this. I think more of these long descriptions of how it works are useful.